### DISCUSSION TITLE
Note-Based HP Drain [1.6-1.7]

### DISCUSSION THREAD

**Discussion by SuxxorState** [KadeArchive/Kade-Engine#2378] [Ideas]
You want to make the enemy drain HP by singing, huh? Neat, let me help you with that.
This is actually quite a simple feat to do, as it's only a few lines of code. So, let's get to it.
Open up **PlayState.hx**.
Here, navigate to around these lines of code (should be around line 3073). Feel free to use the find function (ctrl + f) if need be.

							#if cpp
							if (luaModchart != null)
								luaModchart.executeState('playerTwoSing', [Math.abs(daNote.noteData), Conductor.songPosition]);
							#end

							dad.holdTimer = 0;

							if (SONG.needsVoices)
								vocals.volume = 1;
					}

underneath that final bracket listed above, add in this function--
					if (health > 0.05 && !daNote.isSustainNote)
						health -= 0.04;

and you're done! feel free to tweak the amount on the hp drain (keep the 0.05 just a bit above the 0.04).

(simple, right?)

**Comment by isakube** [comment]
pr this into a guide. you are pogner
